src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int maxSize = 10;
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        int figureType = random.nextInt(5);
+        switch (figureType) {
+            case 0:
+                int side = random.nextInt(maxSize) + 1;
+                String color = colorSupplier.getRandomColor();
+                return new Square(color, side);
```

Comment:
add private methods like getRandomSquare()

Reply:
It seems to me like i did it, but you should check. I'm not sure everything is correct.